### ISSUE TITLE
Create fabric8 add-on

### ISSUE THREAD

**Issue by hferentschik** [minishift/minishift-addons#4]
Moved from https://github.com/minishift/minishift/issues/1034

Lets create an addon so its easy to install fabric8 into minishift from the command line.

**Comment by jstrachan** [comment]
to test out the addon start minishift e.g.:
```
minishift start --vm-driver=xhyve --memory=5000 --cpus=5 --disk-size=50g
```
Then after installing a local copy of the fabric8 addon I was able to apply it via:
```
minishift addon apply --addon-env GITHUB_OAUTH_CLIENT_ID=$GITHUB_OAUTH_CLIENT_ID --addon-env GITHUB_OAUTH_CLIENT_SECRET=$GITHUB_OAUTH_CLIENT_SECRET fabric8
```

**Comment by gbraad** [comment]
@jstrachan can you motivate the amount of assigned vCPUs?

**Comment by jstrachan** [comment]
currently fabric8 hasn't added small cpu limits to all the pods it requires; so we need to tell the VM its got at least 5 CPUs for it to provision everything. Am hoping we can remove that restriction though

**Comment by gbraad** [comment]
Most of the developers will likely run a machine with either 4 (and maybe hopefully 8) cores. This is why I am a little bit concerned. Hope this can be resolved...

**Comment by gbraad** [comment]
@hferentschik do we need a small overview/list about what the add-ons do?

**Comment by jstrachan** [comment]
@gbraad agreed - fabric8 can run fine on a fairly low spec machine that has reasonable memory (5-6Gb for the VM) - its just a temporary limitation that the pods don't yet have low CPU limits on them so that kubernetes doesn't realise it can all really fit fine on a machine with just 4 cores

**Comment by jstrachan** [comment]
incidentally even with a lower cpu count, fabric8 generally starts up fine; its just your jenkins master + builds won't start later on if the number if CPUs is not > 4

**Comment by hrishin** [comment]
Facing the issue with addon for exposecontroller

Logs: 
```I0807 13:09:55.035556       1 exposecontroller.go:46] Using build: '2.3.17'
F0807 13:09:55.075272       1 exposecontroller.go:60] failed to read config file: failed to unmarshal config: yaml: unmarshal errors:
  line 7: cannot unmarshal !!str `${TLS_A...` into bool


[hshinde@localhost ~]$ minishift version 
Minishift Version:
minishift v1.3.1+27ec32a


[hshinde@localhost ~]$ oc version 
oc v1.5.1+7b451fc
kubernetes v1.5.2+43a9be4
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://192.168.42.49:8443
openshift v3.6.0+c4dd4cf
kubernetes v1.6.1+5115d708d7

[hshinde@localhost ~]$ gofabric8 version
gofabric8, version 0.4.138 (branch: 'master', revision: '8c0b80c3')
  build date:       '20170801-15:44:47'
  go version:       '1.8.1'```

**Comment by jstrachan** [comment]
@hrishin could you retry with the latest greatest PR - I just added the missing `HTTP` parameter which should avoid those issues

**Comment by rawlingsj** [comment]
I've also fixed a miss match in the TLS_ACME parameter (https://github.com/fabric8io/fabric8-platform/commit/ab8f0c1dee12bc82cb823b693fa597e6ef159890) so this should work now

**Comment by hrishin** [comment]
expose controller is fixed but init-tenant is not working. 

error logs:
```

{"level":"info","msg":"Configured connection pool max idle 90","time":"2017-08-08 08:28:25"}
--
  | {"level":"info","msg":"Configured connection pool max open 90","time":"2017-08-08 08:28:25"}
  | {"current_version":1,"level":"info","msg":"Current version 1. Nothing to update.","next_version":2,"pkg":"github.com/fabric8-services/fabric8-tenant/migration","time":"2017-08-08 08:28:25"}
  | {"err":"Unknown response:\n\u003chtml\u003e\n  \u003chead\u003e\n    \u003cmeta name=\"viewport\" content=\"width=device-width, initial-scale=1\"\u003e\n\n  \u003cstyle type=\"text/css\"\u003e\n  /*!\n   * Bootstrap v3.3.5 (http://getbootstrap.com)\n   * Copyright 2011-2015 Twitter, Inc.\n   * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)\n   */\n  /*! normalize.css v3.0.3 \| MIT License \| github.com/necolas/normalize.css */\n  html {\n    font-family: sans-serif;\n    -ms-text-size-adjust: 100%;\n    -webkit-text-size-adjust: 100%;\n  }\n  body {\n    margin: 0;\n  }\n  h1 {\n    font-size: 1.7em;\n    font-weight: 400;\n    line-height: 1.3;\n    margin: 0.68em 0;\n  }\n  * {\n    -webkit-box-sizing: border-box;\n    -moz-box-sizing: border-box;\n    box-sizing: border-box;\n  }\n  *:before,\n  *:after {\n    -webkit-box-sizing: border-box;\n    -moz-box-sizing: border-box;\n    box-sizing: border-box;\n  }\n  html {\n    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);\n  }\n  body {\n    font-family: \"Helvetica Neue\", Helvetica, Arial, sans-serif;\n    line-height: 1.66666667;\n    font-size: 13px;\n    color: #333333;\n    background-color: #ffffff;\n    margin: 2em 1em;\n  }\n  p {\n    margin: 0 0 10px;\n    font-size: 13px;\n  }\n  .alert.alert-info {\n    padding: 15px;\n    margin-bottom: 20px;\n    border: 1px solid transparent;\n    background-color: #f5f5f5;\n    border-color: #8b8d8f;\n    color: #363636;\n    margin-top: 30px;\n  }\n  .alert p {\n    padding-left: 35px;\n  }\n  a {\n    color: #0088ce;\n  }\n\n  ul {\n    position: relative;\n    padding-left: 51px;\n  }\n  p.info {\n    position: relative;\n    font-size: 15px;\n    margin-bottom: 10px;\n  }\n  p.info:before, p.info:after {\n    content: \"\";\n    position: absolute;\n    top: 9%;\n    left: 0;\n  }\n  p.info:before {\n    content: \"i\";\n    left: 3px;\n    width: 20px;\n    height: 20px;\n    font-family: serif;\n    font-size: 15px;\n    font-weight: bold;\n    line-height: 21px;\n    text-align: c
  | enter;\n    color: #fff;\n    background: #4d5258;\n    border-radius: 16px;\n  }\n\n  @media (min-width: 768px) {\n    body {\n      margin: 4em 3em;\n    }\n    h1 {\n      font-size: 2.15em;}\n  }\n\n  \u003c/style\u003e\n  \u003c/head\u003e\n  \u003cbody\u003e\n    \u003cdiv\u003e\n      \u003ch1\u003eApplication is not available\u003c/h1\u003e\n      \u003cp\u003eThe application is currently not serving requests at this endpoint. It may not have been started or is still starting.\u003c/p\u003e\n\n      \u003cdiv class=\"alert alert-info\"\u003e\n        \u003cp class=\"info\"\u003e\n          Possible reasons you are seeing this page:\n        \u003c/p\u003e\n        \u003cul\u003e\n          \u003cli\u003e\n            \u003cstrong\u003eThe host doesn't exist.\u003c/strong\u003e\n            Make sure the hostname was typed correctly and that a route matching this hostname exists.\n          \u003c/li\u003e\n          \u003cli\u003e\n            \u003cstrong\u003eThe host exists, but doesn't have a matching path.\u003c/strong\u003e\n            Check if the URL path was typed correctly and that the route was created using the desired path.\n          \u003c/li\u003e\n          \u003cli\u003e\n            \u003cstrong\u003eRoute and path matches, but all pods are down.\u003c/strong\u003e\n            Make sure that the resources exposed by this route (pods, services, deployment configs, etc) have at least one pod running.\n          \u003c/li\u003e\n        \u003c/ul\u003e\n      \u003c/div\u003e\n    \u003c/div\u003e\n  \u003c/body\u003e\n\u003c/html\u003e\n\nGET /auth/realms/fabric8 HTTP/1.1\r\nHost: keycloak-fabric8.192.168.42.16.nip.io\r\nAccept: application/json\r\n\r\n","level":"panic","msg":"failed to parse public token","pid":1,"time":"2017-08-08 08:28:25"}
  | panic: (*logrus.Entry) (0xa1e740,0xc42026adc0)
  |  
  | goroutine 1 [running]:
  | github.com/fabric8-services/fabric8-tenant/vendor/github.com/Sirupsen/logrus.Entry.log(0xc42008e280, 0xc42031c240, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x40e600, ...)
  | /home/jenkins/go/src/github.com/fabric8-services/fabric8-tenant/vendor/github.com/Sirupsen/logrus/entry.go:124 +0x340
  | github.com/fabric8-services/fabric8-tenant/vendor/github.com/Sirupsen/logrus.(*Entry).Panic(0xc42026ad70, 0xc4201618d8, 0x1, 0x1)
  | /home/jenkins/go/src/github.com/fabric8-services/fabric8-tenant/vendor/github.com/Sirupsen/logrus/entry.go:169 +0x11b
  | github.com/fabric8-services/fabric8-tenant/vendor/github.com/Sirupsen/logrus.(*Entry).Panicln(0xc42026ad70, 0xc420161950, 0x1, 0x1)
  | /home/jenkins/go/src/github.com/fabric8-services/fabric8-tenant/vendor/github.com/Sirupsen/logrus/entry.go:264 +0xe0
  | github.com/fabric8-services/fabric8-tenant/vendor/github.com/fabric8-services/fabric8-wit/log.Panic(0x0, 0x0, 0xc42031c180, 0xa4492e, 0x1c, 0x0, 0x0, 0x0)
  | /home/jenkins/go/src/github.com/fabric8-services/fabric8-tenant/vendor/github.com/fabric8-services/fabric8-wit/log/log.go:248 +0x24f
  | main.main()
  | /home/jenkins/go/src/github.com/fabric8-services/fabric8-tenant/main.go:117 +0x1d62
```

**Comment by hrishin** [comment]
Now its got working!

**Comment by jstrachan** [comment]
@hrishin yeah - there's a known issue with starting fabric8 from a cold start; various containers fail as their dependencies are not up yet. 

More details here: 
https://github.com/fabric8io/fabric8-platform/issues/478

we're hoping to add init-containers to delay starting containers until their dependencies are started; to avoid lots of startup errors & restarts which are needless  & just an effect of containers starting in a random order really

**Comment by hferentschik** [comment]
Merged via pull request #5
